fiftyone/core/map/batcher/id_batch.py

```diff
@@ -0,0 +1,72 @@
+"""
+Abstract mapping backend
+
+| Copyright 2017-2025, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+
+import functools
+from typing import List, Optional, TypeVar
+
+import bson
+
+import fiftyone.core.utils as fou
+import fiftyone.core.map.batcher.batch as fomb
+import fiftyone.core.map.batcher.slice_batch as foms
+from fiftyone.core.map.typing import SampleCollection
+
+fov = fou.lazy_import("fiftyone.core.view")
+
+T = TypeVar("T")
+
+
+class SampleIdBatch(fomb.SampleBatch):
+    """Sample batch using ids"""
+
+    @classmethod
+    @functools.cache
+    def get_max_batch_size(cls):
+        """Get max batch size"""
+        return fou.recommend_batch_size_for_value(
+            bson.ObjectId(), max_size=100000
+        )
+
+    @classmethod
+    def split(
+        cls,
+        sample_collection: SampleCollection[T],
+        num_workers: int,
+        batch_size: Optional[int] = None,
+    ) -> List["SampleIdBatch"]:
+        num_workers = max(num_workers, 1)
+
+        ids = sample_collection.values("id")
```

Comment:
On large datasets, this alone could take a long time. Since id is default, end users will experience a hang with no indication of what's going on.

Reply:
Yeah we should utilize the find-hint approach here, for sure.

Either
```python
if is full dataset:
    ids = list(sample_collection._sample_collection.find({}, {"_id": True}, hint={"_id": 1}))
```
Or if we don't wanna use protected stuff, we can make a `fast_ids()` method or something like this.

Reply:
Exposing this as `values(..., unordered=True)` felt natural at first glance to me.

However, adding a kwarg to the public interface like that should really promise that it works for as many possible syntaxes of `values()` as possible. Not just one or two specific things.

Unfortunately I don't think the "fast values" approach that uses `find()` will be able to translate cases like `values(expr)`, because mongo aggregations and mongo find operations support slightly different sets of operations 😢

Reply:
Yeah and find() isn't always fast, just in a few particular cases.

Reply:
yep and `values("id")` and `values("filepath")` are by far the most common values operations, so we should definitely use this trick whenever we can tolerate unordered listing.